src/Command/ErrorFormatter/JsonErrorFormatter.php

```diff
@@ -0,0 +1,48 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use PHPStan\Command\AnalysisResult;
+use Symfony\Component\Console\Style\OutputStyle;
+
+class JsonErrorFormatter implements ErrorFormatter
+{
+
+	public function formatErrors(AnalysisResult $analysisResult, OutputStyle $style): int
+	{
+		$errorsArray = [
+			'totals' => [
+				'errors' => count($analysisResult->getNotFileSpecificErrors()),
+				'file_errors' => count($analysisResult->getFileSpecificErrors()),
+			],
+			'files' => [],
+		];
+
+		foreach ($analysisResult->getFileSpecificErrors() as $fileSpecificError) {
+			if (!array_key_exists($fileSpecificError->getFile(), $errorsArray['files'])) {
+				$errorsArray['files'][$fileSpecificError->getFile()] = [
+					'errors' => 0,
+					'messages' => [],
+				];
+			}
+			$errorsArray['files'][$fileSpecificError->getFile()]['errors']++;
+
+			array_push($errorsArray['files'][$fileSpecificError->getFile()]['messages'], [
+				'message' => $fileSpecificError->getMessage(),
+				'line' => $fileSpecificError->getLine(),
+				'ignorable' => $fileSpecificError->canBeIgnored(),
+			]);
+		}
+
+		$json = json_encode($errorsArray, JSON_UNESCAPED_UNICODE);
```

Comment:
One more thing - please use `Nette\Utils\Json::encode` which handles the errors well - let the exception bubble through (it will result in an internal errors) and return the `if ($json === false) {` condition. Thanks!